core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java

```diff
@@ -152,8 +152,11 @@ private InfrastructureProvisionResult provisionInfrastructure(InfrastructureConf
             return provisionResult;
         } catch (TestGridInfrastructureException e) {
             setTestPlanStatus(testPlan, Status.FAIL);
+            //Destroy created infrastructure if creation failed
+            logger.error("Infrastructure creation failed");
```

Comment:
We have swallowed the exception! And, please do this as the first thing under the catch-block.